src/NATS.Client.Core/NatsConnection.cs

```diff
@@ -84,16 +94,25 @@ public NatsConnection(NatsOpts opts)
             SingleReader = false,
             AllowSynchronousContinuations = false,
         };
+
+        // push consumer events to a channel so handlers can be awaited (also prevents user code from blocking us)
+        _eventChannel = Channel.CreateUnbounded<(NatsEvent, object)>(new UnboundedChannelOptions
+        {
+            AllowSynchronousContinuations = false,
+            SingleWriter = false,
+            SingleReader = true,
+        });
```

Comment:
I'm actually super happy to see this, as in the past it was a hurdle for some heavy consumer cases.